### ISSUE TITLE
AADB2C90085 internal error + Missing Correlation ID

### ISSUE THREAD

**Issue by bpossolo** [AzureAD/microsoft-authentication-library-for-java#402]
Just started getting this error when trying to authenticate via msal4j.
My application is using com.microsoft.azure:msal4j:1.8.0
I tried upgrading to msal 1.9.0 to see if it fixes the problem however no difference.
I can't upgrade to 1.9.1+ yet since it introduces breaking changes in the nimbus-jose library.

The error indicates to please reauthenticate and try again however that does not solve the problem.
Happens on sign-up/sign-in flow when creating new account or logging in to existing account.
I've also tried completely logging out of Azure B2C tenant via the logout url but also makes no difference.

Looks like Azure B2c is returning a null correlation id...
Was there a change in B2C which could be causing this?

```
2021-06-09 09:36:35 INFO  com.microsoft.aad.msal4j.HttpHelper  [Correlation ID: 01f59932-5acc-4c6e-8e64-034963b6684a] Sent (01f59932-5acc-4c6e-8e64-034963b6684a) Correlation Id is not same as received (null).
java.util.concurrent.CompletionException: com.microsoft.aad.msal4j.MsalInteractionRequiredException: AADB2C90085: The service has encountered an internal error. Please reauthenticate and try again.
Correlation ID: fe0fbc52-c507-4e1c-b1dc-198dca4e1190
Timestamp: 2021-06-09 16:36:35Z
	at com.microsoft.aad.msal4j.AuthenticationResultSupplier.get(AuthenticationResultSupplier.java:89)
	at com.microsoft.aad.msal4j.AuthenticationResultSupplier.get(AuthenticationResultSupplier.java:17)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: com.microsoft.aad.msal4j.MsalInteractionRequiredException: AADB2C90085: The service has encountered an internal error. Please reauthenticate and try again.
Correlation ID: fe0fbc52-c507-4e1c-b1dc-198dca4e1190
Timestamp: 2021-06-09 16:36:35Z
	at com.microsoft.aad.msal4j.MsalServiceExceptionFactory.fromHttpResponse(MsalServiceExceptionFactory.java:37)
	at com.microsoft.aad.msal4j.TokenRequestExecutor.createAuthenticationResultFromOauthHttpResponse(TokenRequestExecutor.java:96)
	at com.microsoft.aad.msal4j.TokenRequestExecutor.executeTokenRequest(TokenRequestExecutor.java:37)
	at com.microsoft.aad.msal4j.AbstractClientApplicationBase.acquireTokenCommon(AbstractClientApplicationBase.java:120)
	at com.microsoft.aad.msal4j.AcquireTokenByAuthorizationGrantSupplier.execute(AcquireTokenByAuthorizationGrantSupplier.java:63)
	at com.microsoft.aad.msal4j.AuthenticationResultSupplier.get(AuthenticationResultSupplier.java:59)
	... 8 more
2021-06-09 09:36:35 ERROR c.m.a.m.ConfidentialClientApplication  [Correlation ID: 01f59932-5acc-4c6e-8e64-034963b6684a] Execution of class com.microsoft.aad.msal4j.AcquireTokenByAuthorizationGrantSupplier failed.
	at com.microsoft.aad.msal4j.MsalServiceExceptionFactory.fromHttpResponse(MsalServiceExceptionFactory.java:37)
	at com.microsoft.aad.msal4j.TokenRequestExecutor.createAuthenticationResultFromOauthHttpResponse(TokenRequestExecutor.java:96)
	at com.microsoft.aad.msal4j.TokenRequestExecutor.executeTokenRequest(TokenRequestExecutor.java:37)
	at com.microsoft.aad.msal4j.AbstractClientApplicationBase.acquireTokenCommon(AbstractClientApplicationBase.java:120)
	at com.microsoft.aad.msal4j.AcquireTokenByAuthorizationGrantSupplier.execute(AcquireTokenByAuthorizationGrantSupplier.java:63)
	at com.microsoft.aad.msal4j.AuthenticationResultSupplier.get(AuthenticationResultSupplier.java:59)
	at com.microsoft.aad.msal4j.AuthenticationResultSupplier.get(AuthenticationResultSupplier.java:17)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Comment by bpossolo** [comment]
Turns out the ActiveDirectory B2C confidential Client Secret had expired...

<img width="890" alt="Screen Shot 2021-06-10 at 12 03 04 AM" src="https://user-images.githubusercontent.com/984705/121480100-8f787d80-c97f-11eb-815d-e77da2648846.png">

It would be really helpful if the MSAL sdk returned a useful error message or code for this specific scenario.
I went on a wild goose chase trying to figure this out and debugging the sdk yielded nothing insightful.
It was only when I went digging around the B2C configuration that I realized the Client Secret had expired.
